pkg/tsdb/azuremonitor/aztokenprovider/token_provider.go

```diff
@@ -162,7 +166,7 @@ func (c *clientSecretTokenRetriever) GetCacheKey() string {
 }
 
 func (c *clientSecretTokenRetriever) Init() error {
-	options := &azidentity.ClientSecretCredentialOptions{AuthorityHost: c.authority}
+	options := &azidentity.ClientSecretCredentialOptions{AuthorityHost: azidentity.AuthorityHost(c.authority)}
```

Comment:
Wondering if we should apply any kind of validation 🤔 cc/ @sunker

Reply:
is there any reason you think so?

Reply:
Actually not haha - I guess there's no mechanism exposed to do that kind of validation so probably it's just fair enough to do this kind of type conversion and let it fail if wrong value is provided. Thanks!